processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -500,21 +508,17 @@ public Row apply(Object input)
 
             DateTime timestamp = granularity.toDateTime(((Number) results.next()).longValue());
 
-            Iterator<AggregatorFactory> aggsIter = aggs.iterator();
-
-            Map<String, Object> event = jsonMapper.convertValue(
-                results.next(),
-                new TypeReference<Map<String, Object>>()
-                {
-                }
-            );
+            Map<String, Object> event = Maps.newLinkedHashMap();
+            Iterator<DimensionSpec> dimsIter = dims.iterator();
+            while (dimsIter.hasNext() && results.hasNext()) {
+              final DimensionSpec factory = dimsIter.next();
+              event.put(factory.getDimension(), results.next());
+            }
```

Comment:
If dimsIter or results has ended but the other still has more, is that an error case we need to consider?